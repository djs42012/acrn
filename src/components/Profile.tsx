import React from "react";
import { ImageProps, Image, Pressable, PressableProps } from "react-native";

type PressableWithoutStyleProps = Omit<PressableProps, "style">;
interface ProfileProps extends ImageProps, PressableWithoutStyleProps {
  height: number;
  width: number;
  callback?: () => any;
}

function Profile(props: ProfileProps) {
  const { source, height, width, callback } = props;
  return (
    <Pressable onPress={callback}>
      <Image source={source} style={{ width, height }} />
    </Pressable>
  );
}

export default Profile;
