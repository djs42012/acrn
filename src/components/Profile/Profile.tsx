import React from 'react';
import { ImageProps, Image, Pressable, PressableProps } from 'react-native';

type PressableWithoutStyleProps = Omit<PressableProps, 'style'>;
interface ProfileProps extends ImageProps, PressableWithoutStyleProps {
  height: number;
  width: number;
  withUrlHint?: true;
  profileLabel?: string;
  callback?: () => any;
  imageFlex?: number;
  imageBackgroundColor?: string;
  imageBorderRadius?: number;
  imageBorderColor?: string;
  imagePadding?: number;
  imageMargin?: number;
}

function Profile(props: ProfileProps) {
  const {
    source,
    height,
    width,
    callback,
    imageFlex,
    imageBackgroundColor,
    imageBorderRadius,
    imageBorderColor,
    imagePadding,
    imageMargin,
  } = props;
  return (
    <Pressable
      onPress={callback}
      style={{
        flex: imageFlex,
        backgroundColor: imageBackgroundColor,
        borderRadius: imageBorderRadius,
        borderColor: imageBorderColor,
        padding: imagePadding,
        margin: imageMargin,
        borderWidth: 2,
      }}
    >
      <Image source={source} style={{ width, height }} />
    </Pressable>
  );
}

export default Profile;
