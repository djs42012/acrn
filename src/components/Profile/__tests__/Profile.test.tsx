import React from 'react';
import Profile from '../Profile';
import { render, fireEvent } from '@testing-library/react-native';

describe('Profile test', () => {
  let profile: any;
  const props = {
    height: 100,
    width: 100,
  };
  beforeEach(() => {
    profile = render(<Profile source={0} {...props} />);
    expect(profile).toMatchSnapshot();
  });
  xtest('placeholder', () => {});
});
