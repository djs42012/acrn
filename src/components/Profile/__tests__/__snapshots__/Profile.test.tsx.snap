// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessible Text Input profile has correct width 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Image
    source={0}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;

exports[`Accessible Text Input renders input text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Image
    source={0}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;
